File: star_pattern/star_pattern.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

void repeat(unsigned int count, char *str, bool newline)
{
    if (newline)
    {
        while (count--)
            printf("%s", str);
        printf("%s", "\n");
    }
    else
    {
        while (count--)
            printf("%s", str);
    }
}

int main()
{
    int size = 15;
    int i = 0;
    int iterations = 0;
    while (1)
    {
        iterations++;
        if (iterations >= size)
            break;
        if (iterations <= (int)(size / 2))
        {
            i++;
            repeat(size - i, " ", false);
            repeat(i, "* ", true);
        }
        else
        {
            i--;
            repeat(size - i, " ", false);
            repeat(i, "* ", true);
        }
    }
    return 0;
}

File: star_pattern/star_pattern.rs
fn main(){
    let mut iterations = 0;
    let mut i = 0;
    let size = 15;

    while iterations < size {
        iterations+=1;
        if iterations <= (size / 2) {
            i+=1;
            print!("{}", " ".repeat(size - i));
            println!("{}", "* ".repeat(i));
        } else {
            i-=1;
            print!("{}"," ".repeat(size - i));
            println!("{}","* ".repeat(i));
        }
    }
}